### DISCUSSION TITLE
Paths longer than 260 characters on Windows

### DISCUSSION THREAD

**Discussion by MartyIX** [dotnet/maui#23097] [Q&A] [answered]
Hi,

I hit the issue with long paths quite often when working with MAUI.

An example is:

```
CS0016	Could not write to output file 'D:\maui\artifacts\obj\Maui.Controls.Sample.Sandbox\Debug\net8.0-windows10.0.19041.0\win10-x64\generated\Microsoft.Maui.Controls.SourceGen\Microsoft.Maui.Controls.SourceGen.CodeBehindGenerator\Controls_DataGrid_Filters_TimeSpanFilterPopupWindowContent.xaml.sg.cs' -- 'Could not find a part of the path 'D:\maui\artifacts\obj\Maui.Controls.Sample.Sandbox\Debug\net8.0-windows10.0.19041.0\win10-x64\generated\Microsoft.Maui.Controls.SourceGen\Microsoft.Maui.Controls.SourceGen.CodeBehindGenerator\Controls_DataGrid_Filters_TimeSpanFilterPopupWindowContent.xaml.sg.cs'.'	Maui.Controls.Sample.Sandbox (net8.0-windows10.0.19041.0)	D:\maui\src\Controls\samples\Controls.Sample.Sandbox\CSC	1	
```

It does not state it explicitly but the issue is really that the path to the generated file is too long.

I know I can make the path shorter but it's really getting to the point that it's bothering. Does anyone know how to allow long paths? Or is it not possible at this point?

Thank you

**Comment by dellis1972** [comment, marked as answer]
You should be able to enable "long path support" on your windows machine. See https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry#enable-long-paths-in-windows-10-version-1607-and-later

Hopefully that will help.

**Reply by MartyIX** [reply]
It seems that just changing registry using the following PowerShell command: `New-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Control\FileSystem" -Name "LongPathsEnabled" -Value 1 -PropertyType DWORD -Force` (under an administrator account) as the article recommends helps.

However, to add more details, it appears that I dug my grave with the following setting `<EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>` (removing that might be a good workaround).
